### PR TITLE
Changed to random non-uniform tile creation

### PR DESCRIPTION
**Pull request by max-kov**
The tiles are now added based on how frequently they appear in English language, whereas before every tile had the same chance of being added to the hand.

### CODE REVIEW

**Comment by max-kov** [comment]
Should I cut the frequencies to a separate file? Feels like overkill for one variable

**Comment by harleygn** [comment]
Perhaps make another branch and implement it on there, and we can compare the differences.

File: main.py [hunk 1]
```diff
@@ -156,10 +186,10 @@ def GetScoreForWord(Word, TileDictionary):
 def UpdateAfterAllowedWord(Word, PlayerTiles, PlayerScore, PlayerTilesPlayed, TileDictionary, AllowedWords):
     PlayerTilesPlayed += len(Word)
     for Letter in Word:
-        if PlayerTiles.find(Letter)>=0:
-          PlayerTiles = PlayerTiles.replace(Letter, "", 1)
+        if PlayerTiles.find(Letter) >= 0:
+            PlayerTiles = PlayerTiles.replace(Letter, "", 1)
```
Comment (harleygn) [inline main.py]: Unnecessary extra indent

File: main.py [hunk 2]
```diff
@@ -156,10 +186,10 @@ def GetScoreForWord(Word, TileDictionary):
...
         else:
-          PlayerTiles = PlayerTiles.replace("▊", "", 1)
+            PlayerTiles = PlayerTiles.replace("▊", "", 1)
```
Comment (harleygn) [inline main.py]: Unnecessary extra indent

File: main.py [hunk 3]
```diff
@@ -30,7 +31,36 @@ def Remove(self):
     def Add(self):
         if self._Rear < self._MaxSize - 1:
             TileLibrary = CreateTileLibrary()
-            RandNo = random.randint(0, 26)
+            tile_frequency = [
```
Comment (harleygn) [inline main.py]: Perhaps list these all on the same line? Or at least move the final brace up a line.

**Review by harleygn — COMMENTED** [review]
Check out my comments, awesome solution though.

File: main.py [hunk 4]
```diff
@@ -1,4 +1,5 @@
 import random
```
Comment (harleygn) [inline main.py:1]: This is now redundant and can be removed

File: main.py [hunk 5]
```diff
@@ -1,4 +1,5 @@
...
+import numpy as np
```
Comment (harleygn) [inline main.py]: This must be specified in a pre requisite document somewhere

File: main.py (hunk 1 above)
Comment (harleygn) [inline main.py:162]: Extra indent

File: main.py (hunk 2 above)
Comment (harleygn) [inline main.py:164]: Extra indent

File: main.py (hunk 3 above)
Comment (harleygn) [inline main.py]: Can this be shortened into a single line, even if it runs over a bit? Or at least move the closing brace up a line.

**Review by harleygn — CHANGES_REQUESTED** [review]

File: main.py (hunk 1 above)
Comment (max-kov) [reply main.py:162]: Pycharm refactor, looks like a good change though. This makes the style consistent.

**Pull request merged**